activestorage/lib/active_storage/attached/model.rb

```diff
@@ -130,12 +153,26 @@ def purge_later
 
         after_commit(on: %i[ create update ]) { attachment_changes.delete(name.to_s).try(:upload) }
 
-        ActiveRecord::Reflection.add_attachment_reflection(
-          self,
+        reflection = ActiveRecord::Reflection.create(
+          :has_many_attached,
           name,
-          ActiveRecord::Reflection.create(:has_many_attached, name, nil, { dependent: dependent }, self)
+          nil,
+          { dependent: dependent, service_name: service },
+          self
         )
+        ActiveRecord::Reflection.add_attachment_reflection(self, name, reflection)
       end
+
+      private
+        def validate_service_configuration(association_name, service)
+          return unless service
+
```

Comment:
What do you think about simply doing this?
```
unless ServiceRegistry.fetch(service)
  raise ArgumentError, "Cannot configure service :#{service} for #{name}##{association_name}"
end
```
Also, I think that would also mean you don't need `return unless service` above.

Reply:
Looks like we still need `return unless service` for cases, when `service` is not explicitly configured and we use a default one (default is validated in the [engine](https://github.com/rails/rails/pull/34935/files#diff-fd33c0087b362ffc63058711a96f5245R108))